### PR TITLE
add README in dummy diag publisher

### PR DESCRIPTION
**Pull request by 1222-takeshi**
## Description

Add README in dummy diag publisher.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: system/util/dummy_diag_publisher/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# dummy_diag_publisher
+
+## Purpose
+
+This package outputs a dummy diag for debugging and developing.
```
Comment (kmiya) [inline system/util/dummy_diag_publisher/README.md:5]: IMO: `diag` might be colloquial, so `dummy diagnostic` + _something_ (messages or topics or errors or data?) would be easier to read. In `diagnostic_aggregator/example/README.md`, a sentence `producing dummy diagnostic data` can be found, but I'm not sure which is better.

Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md]: fix "diag" to "diagnostic data"
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/627/commits/48334776cc75e027a44f08084d261f469685032d

File: system/util/dummy_diag_publisher/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+This package outputs a dummy diagnostic data for debugging and developing.
+
+## Inputs / Outputs
+
+### Outputs
+
+| Name               | Type                                    | Description                                                                                                                                                                                |
+| ------------------ | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `/diagnostics`     | `diagnostic_msgs/DiagnosticArray`       | Diagnostics outputs                                                                                                                                                                        |
+| `/diagnostics_agg` | `diagnostic_msgs::msg::DiagnosticArray` | Use aggregated diagnostic information based on [diagnostic_aggregator setting](https://github.com/tier4/autoware.iv/tree/main/system/autoware_error_monitor/config/diagnostic_aggregator). |
```
Comment (tkimura4) [inline system/util/dummy_diag_publisher/README.md:14]: autoware.iv -> AutowareArchitectureProposal.iv

Comment (tkimura4) [reply system/util/dummy_diag_publisher/README.md:14]: (というか、dummy_diagってdiag aggregatorpublishしてますか...?)

File: system/util/dummy_diag_publisher/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+This package outputs a dummy diagnostic data for debugging and developing.
+
+## Inputs / Outputs
+
+### Outputs
+
+| Name               | Type                                    | Description                                                                                                                                                                                |
+| ------------------ | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `/diagnostics`     | `diagnostic_msgs/DiagnosticArray`       | Diagnostics outputs                                                                                                                                                                        |
```
Comment (taikitanaka3) [inline system/util/dummy_diag_publisher/README.md:13]: namespaceと/どちらかに統一したほうがいいかも
```suggestion
| `/diagnostics`     | `diagnostic_msgs::msg::DiagnosticArray`       | Diagnostics outputs                                                                                                                                                                        |
```

File: system/util/dummy_diag_publisher/README.md (hunk 2 above)
Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md:14]: diagnostics_aggとdiagnostics_errは出てなかったです :bow:

File: system/util/dummy_diag_publisher/README.md (hunk 3 above)
Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md]: fix https://github.com/tier4/AutowareArchitectureProposal.iv/pull/627/commits/f8d2cbd17e775c3e57542727749209b59a9460f6

File: system/util/dummy_diag_publisher/README.md (hunk 2 above)
Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md]: remove /diagnostics_agg and /diagnostics_err
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/627/commits/f8d2cbd17e775c3e57542727749209b59a9460f6

**Review by tkimura4 — APPROVED** [review]
LGTM

File: system/util/dummy_diag_publisher/README.md (hunk 2 above)
Comment (taikitanaka3) [reply system/util/dummy_diag_publisher/README.md]: @1222-takeshi 
autoware error monitorのReadmeになってそうです！？
autoware dummy diag publisherはデフォルトのdiag publlisherにarg設定して各エラーをpublishしています。
```
  <include file="$(find-pkg-share dummy_diag_publisher)/launch/dummy_diag_publisher_node.launch.xml">
    <arg name="diag_name" value="velodyne_connection"/>
  </include>
```

File: system/util/dummy_diag_publisher/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| `/diagnostics_agg` | `diagnostic_msgs::msg::DiagnosticArray` | Use aggregated diagnostic information based on [diagnostic_aggregator setting](https://github.com/tier4/autoware.iv/tree/main/system/autoware_error_monitor/config/diagnostic_aggregator). |
+| `/diagnostics_err` | `diagnostic_msgs::msg::DiagnosticArray` | This information is used to visualize HazardStatus.                                                                                                                                        |
```
Comment (taikitanaka3) [inline system/util/dummy_diag_publisher/README.md]: may be unnecessary

Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md]: Yes, /diagnostics_agg and /diagnostics_err have been removed here.
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/627/commits/f8d2cbd17e775c3e57542727749209b59a9460f6

File: system/util/dummy_diag_publisher/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+| -------------- | ---------------------------------------- | ------------------- |
+| `/diagnostics` | `diagnostic_msgs::msgs::DiagnosticArray` | Diagnostics outputs |
+
+## Parameters
+
+### Node Parameters
+
+| Name          | Type | Default Value | Explanation            |
+| ------------- | ---- | ------------- | ---------------------- |
+| `update_rate` | int  | `10`          | Timer callback period. |
```
Comment (taikitanaka3) [inline system/util/dummy_diag_publisher/README.md:21]: @1222-takeshi 
can you add diag name and is_active?
```
      <param name="diag_name" value="$(var diag_name)" />
      <param name="update_rate" value="$(var update_rate)" />
      <param name="is_active" value="$(var is_active)" />
```

diag name is a dummy diag set by dummy diag publisher

Comment (taikitanaka3) [reply system/util/dummy_diag_publisher/README.md:21]: is_activeはTimerCallbackでforce updateするかどうかのフラグっぽいですね。

Comment (kenji-miyake) [reply system/util/dummy_diag_publisher/README.md:21]: 出力を止めたい時用です。
でもROS2だと自動でtimer出力してしまうので、動作周期を1000秒くらいに下げていたと思います。

Comment (1222-takeshi) [reply system/util/dummy_diag_publisher/README.md]: Sorry, I missed those.
Add diag_name and is_active parameters. https://github.com/tier4/AutowareArchitectureProposal.iv/pull/627/commits/8b91302c43918b5e1999fd7f5653b0565bcce26e

Comment (taikitanaka3) [reply system/util/dummy_diag_publisher/README.md]: @kenji-miyake 
> 動作周期を1000秒くらいに下げていた
1000msですかね？
update rate 1[Hz]のほうがよさそうです？

Comment (kenji-miyake) [reply system/util/dummy_diag_publisher/README.md]: 1000秒です。

Comment (taikitanaka3) [reply system/util/dummy_diag_publisher/README.md]: @1222-takeshi 
is_activeをfalseにしても1000秒に一回はpublishされてしまうということらしいのでそのままでも大丈夫です。（注意書きとしてexplanationに書き足すかはおまかせします。）

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Pull request merged**